src/bp-search/classes/class-bp-search-posts.php

```diff
@@ -146,8 +146,10 @@ protected function generate_html( $template_type = '' ) {
 			// lets do a wp_query and generate html for all posts
 			$qry = new WP_Query(
 				array(
-					'post_type' => $this->pt_name,
-					'post__in'  => $post_ids,
+					'post_type' 		=> $this->pt_name,
+					'post__in'  		=> $post_ids,
+					// Override global per page settings and just query with the given IDs
+					'posts_per_page'	=> -1
```

Comment:
@mahedihasannoman I don't think this is the correct solution. if will return all post. We can found count on found posts.